nomad/structs/structs.go

```diff
@@ -10395,6 +10411,56 @@ func (v *Vault) Validate() error {
 	return mErr.ErrorOrNil()
 }
 
+type Secret struct {
+	Name     string
+	Provider string
+	Path     string
+	Config   map[string]any
+}
```

Comment:
You'll likely want an `Equal` method for this for the scheduler. See [the checklist](https://github.com/hashicorp/nomad/blob/main/contributing/checklist-jobspec.md)